### DISCUSSION TITLE
Trade Size with fixed amount in $ ?

### DISCUSSION THREAD

**Discussion by pliciwebstephane** [kernc/backtesting.py#401] [Q&A]
Hello,

Is there a way to open a position with fixed amount, for exemple 100 $ .
Because, using size = 1, is to simulate buying one asset (for sample, if BTC is at 35000 $ , is for buying 1 BTC with 35000 $).

100 $ is 0,002857143 BTC

But if i open a position with size = 0,002857143, it is not 0,002857143 BTC but it is 0.2857143% of my wallet.

Is there a way to simulate a fixed amount of buying ? 

Sorry for my poor english... :)

**Comment by atakanokan** [comment]
Seconded.

https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Order.size states 
```
If size is a value between 0 and 1, it is interpreted as a fraction of current available liquidity (cash plus [Position.pl](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Position.pl) minus used margin). A value greater than or equal to 1 indicates an absolute number of units.
```
which kind of suggests there is no way to trade in equal-dollar positions.

**Reply by kernc** [reply]
Not sure if the notification comes for this Q&A format. See https://github.com/kernc/backtesting.py/discussions/401#discussioncomment-2166017.

**Reply by atakanokan** [reply]
Yep yep, received a notification from the below reply too

**Comment by kernc** [comment]
Buying for fixed $100 is rather easy:
```py
        price = self.data.Close[-1]
        size = 100 // price
        if size:
            self.buy(size=size, ...)
```
In case of Bitcoin, one would first need to [convert that to non-fractional uBTC or satoshis](https://github.com/kernc/backtesting.py/issues/134).

**Reply by atakanokan** [reply]
So the logic is if `trade_dollar_amount // price < 1` then convert asset price to a lower value so that `trade_dollar_amount // price >= 1`? + do the inverse transformation to the `Volume`?

**Reply by kernc** [reply]
Integer division (`//`) will produce only integers (0, 1, ...).

If the price is often high like Bitcoin's, that you might only buy some fraction of units (backtesting.py doesn't support ordering fractional units), convert the whole data to some smaller units like in the example, yes.
